packages/callout/src/Callout.tsx

```diff
@@ -0,0 +1,203 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { css, cx } from '@leafygreen-ui/emotion';
+import { uiColors } from '@leafygreen-ui/palette';
+import Icon, { glyphs } from '@leafygreen-ui/icon';
+
+export const Variant = {
+  Note: 'note',
+  Tip: 'tip',
+  Important: 'important',
+  Warning: 'warning',
+} as const;
+
+export type Variant = typeof Variant[keyof typeof Variant];
+
+const baseStyle = css`
+  border-radius: 6px;
+  box-shadow: 0px 2px 5px 0 ${uiColors.gray.light2};
+  display: flex;
+  overflow: hidden;
+  width: 700px;
+
+  &:before {
+    content: '';
+    position: relative;
+    width: 6px;
+    left: 0px;
+    border-radius: 6px 0px 0px 6px;
+  }
+`;
+
+const headerStyle = css`
+  display: flex;
+  font-family: Akzidenz, ‘Helvetica Neue’, Helvetica, Arial, sans-serif;
+  font-size: 12px;
+  font-weight: bold;
+  letter-spacing: 0.8px;
+  line-height: 16px;
+  padding: 12px 24px 12px 52px;
+  text-transform: uppercase;
+`;
+
+const headerIconStyle = css`
+  margin-left: -32px;
+  position: absolute;
+`;
+
+const titleStyle = css`
+  font-weight: bold;
+  margin: 8px 0;
+`;
+
+const bodyStyle = css`
+  display: flex;
+  flex-direction: column;
+  padding: 16px 24px 20px 52px;
+`;
+
+const bodyTextStyle = css`
+  font-size: 14px;
+  line-height: 20px;
+`;
+
+const headerLabel = {
+  [Variant.Note]: 'Note',
+  [Variant.Tip]: 'Tip',
+  [Variant.Important]: 'Important',
+  [Variant.Warning]: 'Warning',
+};
+
+const headerIcon: Record<Variant, keyof typeof glyphs> = {
+  [Variant.Note]: 'Edit',
+  [Variant.Tip]: 'Bulb',
+  [Variant.Important]: 'InfoWithCircle',
+  [Variant.Warning]: 'Warning',
+};
+
+const colorSets: Record<Variant, ColorSet> = {
+  [Variant.Note]: {
+    background: uiColors.blue.light3,
+    foreground: uiColors.blue.dark2,
+    text: uiColors.blue.dark3,
+  },
+  [Variant.Tip]: {
+    background: '#F3F2FC', // purple light3
+    foreground: '#5A59C6', // purple dark2
+    text: '#2C3A64', // purple dark3
+  },
+  [Variant.Important]: {
+    background: uiColors.yellow.light3,
+    foreground: uiColors.yellow.dark2,
+    text: uiColors.yellow.dark3,
+  },
+  [Variant.Warning]: {
+    background: uiColors.red.light3,
+    foreground: uiColors.red.dark2,
+    text: uiColors.red.dark3,
+  },
+};
+
+interface ColorSet {
+  background: string;
+  foreground: string;
+  text: string;
+}
+
+interface CustomCalloutProps {
+  colorSet: ColorSet;
+  headerIcon?: keyof typeof glyphs;
+  headerLabel: string;
+  title?: string;
+  children: string;
+}
+
+interface CalloutProps
+  extends Omit<CustomCalloutProps, 'colorSet' | 'headerLabel' | 'headerIcon'> {
+  variant: Variant;
+}
+
+function CustomCallout({
```

Comment:
Curious, why is this a separate component? Why is this not embedded into the component itself?

Reply:
My thinking is that similar to breaking up a complex function into smaller functions, I wanted to break up the component into ones with more focused concerns. I would want to write more exhaustive tests for `CustomCallout` and more shallow tests for each variant of `Callout`.

Reply:
Hrm, I understand the sentiment but in this case I'm not sure I see how this allows for more exhaustive testing. I'm happy to hold that judgement until I see the tests, but as it looks now it just looks like we're moving the tests from Callout to CustomCallout. I'm not sure Callout is complex enough to require this decoupling of logic and API.

Reply:
So the idea is to make this unit testable?

I'm not sure this component is complex enough to warrant this separation. IMO it makes it harder to read the code in full and piece together what the resulting rendered code looks like.

The tests you have for the CustomCallout seem simple enough to apply directly to just one component. That also lets us test the entire public API of the component, which we don't really have right now. We moved some of those tests to this inner component.